src/mlpack/methods/ne/gene.hpp

```diff
@@ -0,0 +1,186 @@
+/**
+ * @file gene.hpp
+ * @author Bang Liu
+ *
+ * Definition of the LinkGene and NeuronGene class.
+ */
+#ifndef MLPACK_METHODS_NE_GENE_HPP
+#define MLPACK_METHODS_NE_GENE_HPP
+
+#include <mlpack/core.hpp>
+
+namespace mlpack {
+namespace ne {
+
+/**
+ * This is enumeration of neuron types.
+ */
+enum NeuronType {
+  NONE = 0,
+  INPUT,
+  BIAS,
+  HIDDEN,
+  OUTPUT
+};
+
+/**
+ * This is enumeration of activation function types.
+ */
+enum ActivationFuncType {
+    SIGMOID = 0,
+    TANH,
+    LINEAR,
+    RELU,
+};
+
+/**
+ * This class defines a link gene.
+ */
+class LinkGene {
+ public:
+  // Default constructor.
+  LinkGene() {}
+
+  // Parametric constructor.
+  LinkGene(unsigned int fromNeuronId,
+  	       unsigned int toNeuronId,
+  	       unsigned int innovationId,
+  	       double weight):
+    aFromNeuronId(fromNeuronId),
+    aToNeuronId(toNeuronId),
+    aInnovationId(innovationId),
+    aWeight(weight)
+  {}
+
+  // Copy constructor.
+  LinkGene(const LinkGene& linkGene) {
+  	aFromNeuronId = linkGene.aFromNeuronId;
+  	aToNeuronId = linkGene.aToNeuronId;
+  	aInnovationId = linkGene.aInnovationId;
+  	aWeight = linkGene.aWeight;
+  }
+
+  // Destructor.
+  ~LinkGene() {}
+
+  // Get aFromNeuronId.
+  unsigned int FromNeuronId() const { return aFromNeuronId; }
+
+  // Get aToNeuronId.
+  unsigned int ToNeuronId() const { return aToNeuronId; }
+
+  // Get aInnovationId.
+  unsigned int InnovationId() const { return aInnovationId; }
+
+  // Set aWeight.
+  void Weight(double weight) { aWeight = weight; }
+
```

Comment:
Can you use:

``double& Weight() { return weight; }``

instead of a setter function that takes an argument, to be consistent with the rest of mlpack.